word_cloud_generator/word_cloud_generator.py

```diff
@@ -0,0 +1,307 @@
+import re
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from pathlib import Path
+from sys import argv
+from time import sleep
+from os import path
+from wordcloud import WordCloud
+from typing import Optional
+from dataclasses import dataclass
+from mashumaro import DataClassDictMixin
+
+import json
+import requests
+
+
+@dataclass
+class IssuesFilter(DataClassDictMixin):
+    output: str
+    start_date: str
+    end_date: str
+    word_cloud: str
+    assigned: Optional[str] = None
+    filter_for: Optional[str] = None
+    filter_not: Optional[str] = None
+
+
+def from_user_inputs(**kwargs):
+    """
+    Take the inputs from a argparse and populate a IssuesFilter dataclass and return it
+    :param kwargs: a dictionary of argparse values
+    """
+
+    return IssuesFilter.from_dict(kwargs)
+
+
+def parse_args(inp_args):
+    """
+    Function to parse commandline args
+    :param inp_args: a set of commandline args to parse (dict)
+    :returns: A dictionary of parsed args
+    """
+    # Get arguments passed to the script
+    parser = ArgumentParser(formatter_class=RawDescriptionHelpFormatter)
+
+    parser.add_argument(
+        "-u",
+        "--username",
+        metavar="USERNAME",
+        help="FedID of the user",
+        required=True,
+    )
+    parser.add_argument(
+        "-p",
+        "--password",
+        metavar="PASSWORD",
+        help="Password of the user",
+        required=True,
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        metavar="OUTPUT",
+        help="Directory to create the output files in",
+        default="output",
+    )
+    parser.add_argument(
+        "-s",
+        "--start_date",
+        metavar="START_DATE",
+        help="Date to get issues from",
+        default=datetime.now().strftime("%Y-%m-%d"),
```

Comment:
For readability can this expression be put into a variable first?
```suggestion
        default=someVarContainingExpression,
```